### PR TITLE
Paste curl modal

### LINKED ISSUE #6338: Feature request: Re-add paste CURL on URL bar
### Expected Behavior

When I past a curl command (copied from Chrome with 'copy as CURL) I expect Insomnia to parse it and I can replay the chrome call in Insomnia:

This has been working forever and is extremely usefull to ne,

### Actual Behavior

Gives me an error

Screenshot from 2023-08-20 09-40-32

### Reproduction Steps

Past this:

curl 'https://jsonplaceholder.typicode.com/posts' \
  -H 'authority: jsonplaceholder.typicode.com' \
  -H 'accept: */*' \
  -H 'accept-language: en-US,en;q=0.9,nl;q=0.8' \
  -H 'cache-control: no-cache' \
  -H 'content-type: application/json; charset=UTF-8' \
  -H 'origin: http://localhost:3001' \
  -H 'pragma: no-cache' \
  -H 'referer: http://localhost:3001/' \
  -H 'sec-ch-ua: "Not/A)Brand";v="99", "Google Chrome";v="115", "Chromium";v="115"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: cross-site' \
  -H 'user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36' \
  --data-raw '{"title":"foo","body":"bar","userId":1}' \
  --compressed

In the address bar of insomnia

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

_No response_

### Insomnia Version

Version: Insomnia 2023.5.5 Build date: 8/18/2023 OS: Linux x64 6.2.0-26-generic Electron: 25.2.0 Node: 18.15.0 Node ABI: 116 V8: 11.4.183.23-electron.0 Architecture: x64

### What operating system are you using?

Ubuntu

### Operating System Version

Linux ubuntu 6.2.0-26-generic #26~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Jul 13 16:27:29 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

### Installation method

SNAP

### Last Known Working Insomnia version

The version

### PR DESCRIPTION
**Pull request by jackkav**
changelog(Fixes): Re-added a way to paste curl requests directly into URL bar.

fixes: #6338

- [x] cancel change to url bar when curl paste detected
- [x] continue to single line mutliline pastes when not curl requests
- [x] always create new request
- [x] add from curl button icon
- [x] add from curl to folder and accept parentId assignment

### CODE REVIEW

**Review by filfreire — APPROVED** [review]
@jackkav there might be a small bug when pasting some curls they are incorrectly picked up as invalid, example:

```
curl --request POST \
  --url 'http://mockbin.org/bin/c24b72a2-1aed-48a3-be2d-b6122e855333?foo=bar&foo=baz' \
  --header 'accept: application/json' \
  --header 'content-type: application/x-www-form-urlencoded' \
  --cookie 'foo=bar; bar=baz' \
  --data foo=bar \
  --data bar=baz
  ```
  
  shows 
imagem

on the bottom. But if I add a space somewhere on the curl text in the modal, it becomes valid

imagem

**Comment by jackkav** [comment]
@filfreire that's correct the validation check should be happening on paste as well as change. Could you give it a try for me please?

**Comment by filfreire** [comment]
@jackkav seems the problem is the isValid and the req start as `null` / empty when the modal is first drawn after it's open

**Review by gatzjames — APPROVED** [review]

File: packages/insomnia/src/ui/components/panes/request-pane.tsx [hunk 1]
```diff
@@ -62,20 +61,24 @@ interface Props {
...
 }) => {
   const { activeRequest, activeRequestMeta } = useRouteLoaderData('request/:requestId') as RequestLoaderData;
   const { workspaceId, requestId } = useParams() as { organizationId: string; projectId: string; workspaceId: string; requestId: string };
   const patchSettings = useSettingsPatcher();
   const [isRequestSettingsModalOpen, setIsRequestSettingsModalOpen] =
     useState(false);
+  const patchRequest = useRequestPatcher();
 
-  const handleImportQueryFromUrl = useCallback(() => {
+  useState(false);
```
Comment (marckong) [inline packages/insomnia/src/ui/components/panes/request-pane.tsx:80]: @jackkav I am wondering if there was a reason for this?

**Comment by krishna-atkalikar** [comment]
instead of _always create new request_, is it possible to modify current request.

**Pull request merged**